remote/remote_test.go

```diff
@@ -168,3 +171,84 @@ func doTestWatch(t *testing.T, sm subnet.Manager) {
 		t.Errorf("WatchSubnet produced wrong subnet: expected %s, got %s", l.Key(), evt.Lease.Key())
 	}
 }
+
+func doTestWatchNetworks(t *testing.T, sm subnet.Manager, serverRegistry *subnet.MockSubnetRegistry) {
+	ctx, cancel := context.WithCancel(context.Background())
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	defer func() {
+		cancel()
+		wg.Wait()
+	}()
+
+	// Because remote clients are not allowed to add/remove networks, there
+	// is no server API for these operations.  But testing network watches
+	// still requires this, and that must happen directly in the server's
+	// registry.  That can cause race conditions if the WatchNetworks
+	// goroutine hasn't been run yet by the time CreateNetwork is called,
+	// so testing operations that touch the server registry must be deferred
+	// a short time to ensure that the server is listening when they happen.
+
+	events := make(chan []subnet.Event)
+	go func() {
+		subnet.WatchNetworks(ctx, sm, events)
+		wg.Done()
+	}()
+
+	// skip over the initial snapshot
+	<-events
+
+	expectedNetname := "foobar"
+	go func() {
```

Comment:
Why a separate goroutine? The current goroutine is just going to block on line 214 anyway.